src/callbacks/onLayout/augmentOtherControls/timeScaleChange.js

```diff
@@ -9,28 +10,15 @@ export default function timeScaleChange(dropdown, d) {
         .select('option:checked')
         .text();
     syncTimeScaleSettings(this.config);
+    this.config.time_range =
+        this.config.time_scale === 'day' ? this.config.day_range : this.config.date_range;
 
     //Update ID timeline time scale settings
     this.IDtimeline.config.time_scale = this.config.time_scale;
     syncTimeScaleSettings(this.IDtimeline.config);
 
-    //Update listing time scale settings
-    this.listing.config.cols = this.listing.config.cols.map(col => {
-        if (col === this.config.stdy_col) col = this.config.stdt_col;
-        else if (col === this.config.stdt_col) col = this.config.stdy_col;
-        if (col === this.config.endy_col) col = this.config.endt_col;
-        else if (col === this.config.endt_col) col = this.config.endy_col;
-
-        return col;
-    });
-    this.listing.config.headers = this.listing.config.headers.map(header => {
-        if (header === 'Start Day') header = 'Start Date';
-        else if (header === 'Start Date') header = 'Start Day';
-        if (header === 'Stop Day') header = 'Stop Date';
-        else if (header === 'Stop Date') header = 'Stop Day';
-
-        return header;
-    });
+    //Remove records without time data.
+    cleanData.call(this);
```

Comment:
Lets say there are 10 missing days and 20 missing dates in a 100 record data set (no overlap). Will there be 90 records in one view, and 80 records in the other - or will there be 70 records in both views? Or some other configuration?

Reply:
Let's add a test case that explicitly spells out (and then tests) the expected behavior here.

Reply:
cleanData() is called after updating the time scale so it removes missing values in that variable from the initial raw data.

Reply:
Gotcha - so in the case above we're expecting 70 records in both views?

Reply:
No, 90 day records and 80 date records.